### PR TITLE
improve modal styles + add a customisation sample

### PR DESCRIPTION
**Pull request by tobi-or-not-tobi**

### CODE REVIEW

File: projects/storefrontstyles/scss/components/product/list/_facet-list.scss [hunk 1]
```diff
@@ -1,3 +1,52 @@
+%cx-dialog {
```
Comment (LTiger14) [inline projects/storefrontstyles/scss/components/product/list/_facet-list.scss:5]: This is a good direction. We should have a global classes or placeholders to use like you did.

File: projects/storefrontstyles/scss/components/product/list/_facet-list.scss [hunk 2]
```diff
@@ -1,3 +1,52 @@
...
+
+  > .inner {
+    // width: 100vw;
+    // height: 100vh;
+    transform: translate3d(0, 100%, 0);
+  }
+}
+
+%cx-overlay {
+  @extend %cx-modal;
```
Comment (LTiger14) [inline projects/storefrontstyles/scss/components/product/list/_facet-list.scss:43]: I guess you could have the overlay as the dialog background like you did or you could have a different element that with this style.

File: projects/storefrontstyles/scss/components/product/list/_facet-list.scss [hunk 3]
```diff
@@ -12,24 +61,21 @@
...
+        color: var(--cx-color-primary);
+      }
+    }
+  }
 
-    &.active {
-      transform: translate3d(0, 0, 0);
+  &.dialog {
+    @extend %cx-modal;
+    > .inner {
```
Comment (LTiger14) [inline projects/storefrontstyles/scss/components/product/list/_facet-list.scss:84]: I guess the width should be dependent on the screen size

File: projects/storefrontapp/src/styles.scss [hunk 4]
```diff
@@ -1,3 +1,17 @@
 // Spartacus Application styles
 
 @import 'storefrontstyles/index';
+
+cx-product-facet-navigation {
+  button.dialog-trigger {
```
Comment (LTiger14) [inline projects/storefrontapp/src/styles.scss:6]: Why did you need put this here?

File: projects/storefrontstyles/scss/components/product/list/_facet-list.scss (hunk 1 above)
Comment (LTiger14) [inline projects/storefrontstyles/scss/components/product/list/_facet-list.scss:5]: One small thing, why is the close button on the dialog only visible in mobile/tablet?

File: projects/storefrontapp/src/styles.scss (hunk 4 above)
Comment (tobi-or-not-tobi) [reply projects/storefrontapp/src/styles.scss:6]: Just to demo that we can influence the experience to force the dialog by showing the button.

File: projects/storefrontstyles/scss/components/product/list/_facet-list.scss (hunk 1 above)
Comment (tobi-or-not-tobi) [reply projects/storefrontstyles/scss/components/product/list/_facet-list.scss:5]: in inline mode, you can't close the facets. It's always there ;)

**Comment by tobi-or-not-tobi** [comment]
closing as we already merged the target PR
